pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -249,7 +249,13 @@ func (c *Controller) GetService(hostname model.Hostname) (*model.Service, error)
 	return svc, nil
 }
 
-// serviceByKey retrieves a service by name and namespace
+// GetServiceNamespace retrieves the namespace from the service.
+func (c *Controller) GetServiceNamespace(service *model.Service) (string, error) {
+	_, ns, err := parseHostname(service.Hostname)
```

Comment:
This is incorrect. If you look at the implementation of parse hostname, that just takes first two words of the service, no matter whether it belongs or not to the register.

Reply:
Can you please just return both values "name" and "namespace"?
Maybe call this function `GetServiceNameNamespace` and return two strings?

Reply:
Sure, I'm fine with either way. Could you suggest the default service value on other platforms?
Also @rshriram any thoughts on this?

Reply:
You can set to empty string if it's not possible to determine. We'll simply not provide the attribute to mixer in that case. You probably don't even error value, if you can first check with `HasService`.

Reply:
The idea behind this function is to not parse the string. It’s to lookup the service from some table and return the namespace associated with the service - this namespace is an istio namespace and not a kubernetes namespace. foo.com could belong to bar namespace for all we care. If you need kubernetes specific semantics in the mixer plugin (eg parsing the short name of a service which doesn’t really work once you move off k8s, you are better off encapsulating that functionality there instead of leaking it go every other plugin via the model.